### ISSUE TITLE
[BUG][Block 657887]  `Native execution error: high !> low\n"`

### ISSUE THREAD

**Issue by pefontana** [lambdaclass/cairo_native#738]
When executing Block 657887 with starknet-replay, we get this error in certain transactions 
```
Native execution error: Native execution error: high !> low\n"
```
This error doesn't occur when we execute the block with only the cairo VM

transactions hash that triggers the error:
* 0x5ba52f32a6a5add8affdd8c7caece321fe76c76e036c49b40db227154916353
* 0x75c5631aefd35cc4f7bf5f245f10a726336219f35c647b3434353722bdcc23b
* 0x47679df0e8395d1e8ea06a9d3730e744a1ab1ad9d68d4d6fb288bbf4b1b6e37
* 0x3ca02cfa0ef0e9d1c0a940008730e0a3a38c2a7ec43bf60088ab50dab01abb7

PD. It is provable that this bug is related to #727 #714

**Comment by FrancoGiachetta** [comment]
Running again block 657887 resulted in: 
- Native: Failed while executing tx `0x7e825719940feba6fbc91d5a1b5c6b7bc77f6a980b5c879123eb135eb4df624` with message `Killed`, though the trasactions that trigger the error are passing now (the same as the vm)

**Comment by pefontana** [comment]
Fixed with #787 and #763
